packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -867,6 +868,46 @@ function deferHiddenOffscreenComponent(
 // fork the function.
 const updateLegacyHiddenComponent = updateOffscreenComponent;
 
+function updateActivityComponent(
+  current: null | Fiber,
+  workInProgress: Fiber,
+  renderLanes: Lanes,
+) {
+  const nextProps = workInProgress.pendingProps;
+  const nextChildren = nextProps.children;
+  const nextMode = nextProps.mode;
+  const mode = workInProgress.mode;
+  const offscreenChildProps: OffscreenProps = {
+    mode: nextMode,
+    children: nextChildren,
+  };
+
+  if (current === null) {
+    const primaryChildFragment = mountWorkInProgressOffscreenFiber(
+      offscreenChildProps,
+      mode,
+      renderLanes,
+    );
+    primaryChildFragment.ref = workInProgress.ref;
```

Comment:
@sebmarkbage currently I'm just forwarding the `ref` to the Offscreen element child, and not changing how manual mode works. As a follow up, should we make it so the `ref` is only exposed to the Activity element, which finds the Offscreen child to toggle visibility?

Reply:
Yes.

Reply:
Removed manual mode in https://github.com/facebook/react/pull/32645